fish-rust/src/env_dispatch.rs

```diff
@@ -0,0 +1,809 @@
+            term_support,
+            "256-color support:",
+            max_colors,
+            "per termcap/terminfo entry for",
+            term
+        );
+    }
+
+    if let Some(fish_term24bit) = vars.get(L!("fish_term24bit")).map(|v| v.as_string()) {
+        // $fish_term24bit
+        supports_24bit = crate::wcstringutil::bool_from_string(&fish_term24bit);
+        FLOGF!(
+            term_support,
+            "$fish_term24bit preference: 24-bit color",
+            if supports_24bit {
+                "enabled"
+            } else {
+                "disabled"
+            }
+        );
+    } else if vars.get(L!("STY")).is_some() || term.starts_with(L!("eterm")) {
+        // Screen and emacs' ansi-term swallow true-color sequences, so we ignore them unless
+        // force-enabled.
+        supports_24bit = false;
+        FLOGF!(
+            term_support,
+            "True-color support: disabled for eterm/screen"
+        );
+    } else if max_colors.unwrap_or(0) > 32767 {
+        // $TERM wins, xterm-direct reports 32767 colors and we assume that's the minimum as xterm
+        // is weird when it comes to color.
+        supports_24bit = true;
+        FLOGF!(
+            term_support,
+            "True-color support: enabled per termcap/terminfo for",
+            term,
+            "with",
+            max_colors.unwrap(),
+            "colors"
+        );
+    } else if let Some(ct) = vars.get(L!("COLORTERM")).map(|v| v.as_string()) {
+        // If someone sets $COLORTERM, that's the sort of color they want.
+        if ct == L!("truecolor") || ct == L!("24bit") {
+            supports_24bit = true;
+        }
+        FLOGF!(
+            term_support,
+            "True-color support",
+            if supports_24bit {
+                "enabled"
+            } else {
+                "disabled"
+            },
+            "per $COLORTERM",
+            ct
+        );
+    } else if vars.get(L!("KONSOLE_VERSION")).is_some()
+        || vars.get(L!("KONSOLE_PROFILE_NAME")).is_some()
+    {
+        // All Konsole versions that use $KONSOLE_VERSION are new enough to support this, so no
+        // check is needed.
+        supports_24bit = true;
+        FLOGF!(term_support, "True-color support: enabled for Konsole");
+    } else if let Some(it) = vars.get(L!("ITERM_SESSION_ID")).map(|v| v.as_string()) {
+        // Supporting versions of iTerm include a colon here.
+        // We assume that if this is iTerm it can't also be st, so having this check inside is okay.
+        if !it.contains(':') {
+            supports_24bit = true;
+            FLOGF!(term_support, "True-color support: enabled for iTerm");
+        }
+    } else if term.starts_with("st-") {
+        supports_24bit = true;
+        FLOGF!(term_support, "True-color support: enabling for st");
+    } else if let Some(vte) = vars.get(L!("VTE_VERSION")).map(|v| v.as_string()) {
+        if fish_wcstoi(vte.chars()).unwrap_or(0) > 3600 {
```

Comment:
This was `fish_wcstod` not `fish_wcstoi`

Reply:
I'd actually like to change it to `u32`'s `FromStr` impl (`string::parse()`) with [the same rational as above](https://github.com/fish-shell/fish-shell/pull/9786#discussion_r1193220614) as it seems the type of VTE_VERSION we are interested in (as there seem to be many) are the ones with higher integral values (in the hundreds or thousands) and not a lower program version like "3.2.0" so that would suffice.